rpcs3/Emu/System.cpp

```diff
@@ -2497,7 +2497,13 @@ void Emulator::GracefulShutdown(bool allow_autoexit, bool async_op, bool savesta
 		for (u32 i = 100; i < 140; i++)
 		{
 			std::this_thread::sleep_for(50ms);
-			Resume(); // TODO: Prevent pausing by other threads while in this loop
+
+			// TODO: Prevent pausing by other threads while in this loop
+			CallFromMainThread([this]()
```

Comment:
why is this async?

Reply:
This is a bit timing dependant, better to not block the thread and affect it. It's not required to be resumed right after.

Reply:
std::thread cannot execute BlockingCallFromMainThread which is present inside Resume().